### PR TITLE
Encapsulate Stepper, Planner, Endstops in singleton classes

### PR DESCRIPTION
**Pull request by thinkyhead**
Improving the semantics of the codebase with singletons.

This PR leaves the code "intact", but:
- Wraps the stepper as a singleton class named `Stepper`.
- Instantiates a `Stepper` named `stepper`.
- Wraps the endstops manager as a singleton class named `Endstops`.
- Instantiates an `Endstops` named `endstops`.
- Wraps the planner as a singleton class named `Planner`.
- Instantiates a `Planner` named `planner`.
- Removes `st_` and `plan_` prefixes from the method names.
- Modifies code to use the `stepper.`, `planner.`, and `endstops.` objects.
- Some other minor changes to facilitate the implementation.

Compiles with various options, but we'll see how Travis likes it. The first time I installed this on my machine it _just worked_. Not too surprising, as this is almost entirely a stylistic change.

It should be noted that a bug is fixed here also in the process:
- `dock_sled` was disabling the probe when undocked, and enabling it when docked. That is now reversed. I may patch that separately.

### CODE REVIEW

**Comment by thinkyhead** [comment]
Travis CI has been showing me the symbols I missed and I've been patching them up. But I have been called away so I will have to pick this up later!

**Comment by thinkyhead** [comment]
How can I coax the compiler into inlining the `Stepper::isr` function in the stepper ISR? Do I have to make it `FORCE_INLINE` it in `stepper.h`? It would be good to save the cost of a method call.

**Comment by thinkyhead** [comment]
This finally passes all the checks in `.travis.yml`.

I have a `Temperature` singleton ready to go as a set of commits on top of this.

As usual, please look hard for any accidental typos or logical errors.

File: Marlin/stepper.cpp [hunk 1]
```diff
@@ -1256,10 +866,14 @@ void quickStop() {
...
 
+/**
+ * Software-controlled Stepper Motor Current
+ */
+
 #if HAS_DIGIPOTSS
 
   // From Arduino DigitalPotControl example
-  void digitalPotWrite(int address, int value) {
+  Stepper::void digitalPotWrite(int address, int value) {
```
Comment (ghost) [inline Marlin/stepper.cpp]: Is `void Stepper::digitalPotWrite(int address, int value) {` right? ([same comment in PR #3643](https://github.com/MarlinFirmware/Marlin/pull/3643#discussion_r61534823))

**Pull request merged**